src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -103,6 +106,35 @@ public Location getLocation(String id){
     Location location = new Location(id, name, lat, lng);
     return location;
   }
+
+  public List<LocationServlet.LocationCount> getLocationCount() {
+    Query query = new Query("Message").setFilter(
+      new Query.FilterPredicate("location_id", FilterOperator.NOT_EQUAL, null));
+    PreparedQuery results = datastore.prepare(query);
+    HashMap<String, Integer> count = new HashMap<>();
+    for(Entity entity: results.asIterable()) {
+      String id = (String)entity.getProperty("location_id");
```

Comment:
Will "count" stores a key as "null" if there's no location_id in the entity?

Reply:
Query filters those messages without location so it doesn't store null.

Reply:
Ah, I see! thanks